### PR TITLE
fix(core): combobox handle IME input correctly to prevent accidental deletion of characters

### PR DESCRIPTION
**Pull request by khotcholava**
fix(core): handle IME input correctly to prevent accidental deletion of characters

closes [#12232](https://github.com/SAP/fundamental-ngx/issues/12232)

## Description
- Added handling for `compositionstart` and `compositionend` events to properly manage IME input (e.g., Chinese, Japanese).
- Introduced `isComposing` flag to bypass autocomplete logic during IME composition.
- Updated `inputText` synchronization after IME input ends to ensure it matches the finalized value.
- Prevented unwanted modifications of input value when using IME, addressing the issue where typing and deleting partially composed characters could remove the last English character.

### CODE REVIEW

File: libs/cdk/utils/directives/auto-complete/auto-complete.directive.ts [hunk 1]
```diff
@@ -57,6 +57,9 @@ export class AutoCompleteDirective {
     /** @hidden */
     private lastKeyUpEvent: KeyboardEvent;
 
+    /** @hidden */
+    private isComposing = false;
```
Comment (InnaAtanasova) [inline libs/cdk/utils/directives/auto-complete/auto-complete.directive.ts:61]: We usually prefix the private members with underscore. And I know the property above is private and not prefixed. We missed it during the review.

File: libs/cdk/utils/directives/auto-complete/auto-complete.directive.ts [hunk 2]
```diff
@@ -72,6 +75,12 @@ export class AutoCompleteDirective {
          */
         this._zone.runOutsideAngular(() => {
             const keyupEvent = fromEvent<KeyboardEvent>(this._elementRef.nativeElement, 'keyup');
+            const compositionStartEvent = fromEvent<CompositionEvent>(
+                this._elementRef.nativeElement,
+                'compositionstart'
```
Comment (InnaAtanasova) [inline libs/cdk/utils/directives/auto-complete/auto-complete.directive.ts:80]: Screenshot 2024-09-03 at 1 47 09 PM
Screenshot 2024-09-03 at 1 47 00 PM

'compositionstart' and 'compositionend' are experimental technology and not supported in all browsers:
https://developer.mozilla.org/en-US/docs/Web/API/EditContext/compositionstart_event

Comment (khotcholava) [reply libs/cdk/utils/directives/auto-complete/auto-complete.directive.ts:80]: Hi @InnaAtanasova, I found this [MDN article on the `compositionstart` event](https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionstart_event), which mentions that it is supported by all major browsers. I also tested it on Safari, Edge, Firefox, and Chrome, and it worked without any issues.

**Review by dpavlenishvili — APPROVED** [review]

**Comment by droshev** [comment]
@khotcholava can you investigate what is wrong with CI?
